packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js

```diff
@@ -30,31 +64,96 @@ const OrderHistoryPage = props => {
         });
     }, [orders]);
 
-    if (isLoadingWithoutData) {
-        return fullPageLoadingIndicator;
-    }
-
-    let pageContents;
-    if (!orders.length) {
-        pageContents = (
-            <h3 className={classes.emptyHistoryMessage}>
-                {EMPTY_DATA_MESSAGE}
-            </h3>
-        );
-    } else {
-        pageContents = (
-            <ul className={classes.orderHistoryTable}>{orderRows}</ul>
-        );
-    }
+    const pageContents = useMemo(() => {
+        if (!isBackgroundLoading && searchText && !orders.length) {
+            return (
+                <h3 className={classes.emptyHistoryMessage}>
+                    <FormattedMessage
+                        id={'orderHistoryPage.invalidOrderNumber'}
+                        defaultMessage={`Order "${searchText}" was not found.`}
+                        values={{
+                            number: searchText
+                        }}
+                    />
+                </h3>
+            );
+        } else if (!isBackgroundLoading && !orders.length) {
+            return (
+                <h3 className={classes.emptyHistoryMessage}>
+                    <FormattedMessage
+                        id={'orderHistoryPage.emptyDataMessage'}
+                        defaultMessage={"You don't have any orders yet."}
+                    />
+                </h3>
+            );
+        } else {
+            return <ul className={classes.orderHistoryTable}>{orderRows}</ul>;
+        }
+    }, [
+        classes.emptyHistoryMessage,
+        classes.orderHistoryTable,
+        isBackgroundLoading,
+        orderRows,
+        orders,
+        searchText
+    ]);
 
     // STORE_NAME is injected by Webpack at build time.
     const title = `${PAGE_TITLE} - ${STORE_NAME}`;
 
+    const resetButton = searchText ? (
+        <Trigger action={resetForm}>{clearIcon}</Trigger>
+    ) : null;
+
+    const submitIcon = (
+        <Icon
+            src={SubmitIcon}
+            size={24}
+            classes={{
+                icon: classes.submitIcon
+            }}
+        />
+    );
+
+    useEffect(() => {
+        if (errorMessage) {
+            addToast({
+                type: 'error',
+                icon: errorIcon,
+                message: errorMessage,
+                dismissable: true,
+                timeout: 10000
+            });
+        }
+    }, [addToast, errorMessage]);
+
+    if (isLoadingWithoutData) {
+        return fullPageLoadingIndicator;
+    }
+
     return (
         <OrderHistoryContextProvider>
             <div className={classes.root}>
                 <Title>{title}</Title>
                 <h1 className={classes.heading}>{PAGE_TITLE}</h1>
+                <div className={classes.search}>
```

Comment:
Something like this:
```suggestion
                <Form onSubmit={handleSubmit} className={classes.search}>
```

Reply:
Nice idea @tjwiebell, Unfortunately, the problem with this solution is the talon won't have access to the `searchText`. The talon needs `searchText` and so does the component. We do some if-else checking in the component that needs the `searchText` from the talon.

Reply:
I've encountered the chicken/egg issue with form context and using those hooks from the parent talon, so understand what you mean. I'll try my suggestion in scope of the pagination ticket, and won't hold this one up.